substrate/primitives/arithmetic/src/helpers_128bit.rs

```diff
@@ -183,21 +183,35 @@ mod double128 {
 }
 
 /// Returns `a * b / c` (wrapping to 128 bits) or `None` in the case of
-/// overflow.
+/// overflow or div by zero.
 pub const fn multiply_by_rational_with_rounding(
 	a: u128,
 	b: u128,
 	c: u128,
 	r: Rounding,
 ) -> Option<u128> {
+	match checked_multiply_by_rational_with_rounding(a, b, c, r) {
```

Comment:
this is the way I found to call it as `unwrap()` and `expect()` are not expected to be inside a `const fn`

Reply:
Yep AFAIK there is no better way for this.